core-bundle/src/Translation/DataCollectorTranslator.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Translation;
+
+use Symfony\Component\Translation\DataCollectorTranslator as SymfonyDataCollectorTranslator;
+use Symfony\Component\Translation\MessageCatalogueInterface;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Symfony\Component\Translation\TranslatorInterface as LegacyTranslatorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class DataCollectorTranslator extends SymfonyDataCollectorTranslator
+{
+    /**
+     * @var TranslatorInterface&TranslatorBagInterface&LegacyTranslatorInterface
+     */
+    private $translator;
+
+    private $messages = [];
+
+    public function __construct($translator)
```

Comment:
PhpStan throws an error here:

```
Property                                                                 
         Contao\CoreBundle\Translation\DataCollectorTranslator::$translator       
         (Symfony\Component\Translation\TranslatorBagInterface&Symfony\Component  
         \Translation\TranslatorInterface&Symfony\Contracts\Translation\Translat  
         orInterface) does not accept                                             
         Symfony\Contracts\Translation\TranslatorInterface.
```

Maybe we have to add a phpDoc comment?

```suggestion
    /**
     * @param TranslatorInterface&TranslatorBagInterface&LegacyTranslatorInterface $translator
     */
    public function __construct($translator)
```